hazelcast/src/main/java/com/hazelcast/query/impl/Index.java

```diff
@@ -125,4 +127,27 @@
      */
     void destroy();
 
+    /**
+     * Identifies an original source of an index operation.
+     * <p>
+     * Required for the index stats tracking to ignore index operations
+     * initiated internally by Hazelcast. We can't achieve the same behaviour
+     * on the pure stats level, e.g. by turning stats off during a partition
+     * migration, since global indexes and their stats are shared across
+     * partitions.
+     */
+    enum OperationSource {
+        /**
+         * Indicates that an index operation was initiated by a user; for
+         * instance, as a result of a new map entry insertion.
+         */
+        User,
```

Comment:
minor: convention for enum literals is to have all caps names

Reply:
fixed